### PR TITLE
FleCSI updates

### PR DESCRIPTION
**Pull request by rbberger**
- deprecate 1.4
- simplify `hdf5` variant logic, since there is an explicit conflict for `backend=hpx`

### CODE REVIEW

**Review by tldahlgren — APPROVED** [review]
LGTM.  Should I wait for @ktsai7 or is this ready to merge (once all required checks pass)?

**Comment by rbberger** [comment]
@tldahlgren this is ready for merge.

**Pull request merged**
